src/components/table/table.test.tsx

```diff
@@ -376,6 +376,77 @@ describe('Table Component', () => {
       const afterHoverCheckboxes = screen.queryAllByTestId('checkbox');
       expect(afterHoverCheckboxes.length).toBeGreaterThanOrEqual(initialCount);
     });
+
+    describe('Select All Checkbox Visibility', () => {
+      it('renders header checkbox when isSelectAllCheckboxAlwaysVisible is true and data exists (even without selection)', () => {
+        const { container } = render(
+          <Table
+            {...defaultProps}
+            selectable={true}
+            selectedRowIds={[]}
+            isSelectAllCheckboxAlwaysVisible={true}
+          />,
+        );
+
+        const checkbox = container.querySelector(
+          '[class*="table-header"] [data-testid="checkbox"]',
+        );
+
+        expect(checkbox).toBeInTheDocument();
+      });
+
+      it('does NOT render header checkbox when isSelectAllCheckboxAlwaysVisible is true but data is EMPTY', () => {
```

Comment:
No need to use capital case
```suggestion
      it('does not render header checkbox when isSelectAllCheckboxAlwaysVisible is true but data is empty', () => {
```